graphql_compiler/compiler/ir_lowering_match.py

```diff
@@ -445,71 +447,81 @@ def remove_backtrack_blocks_from_fold(folded_ir_blocks):
     return new_folded_ir_blocks
 
 
-def convert_optional_traversals_to_compound_match_query(match_query):
+def _filter_local_field_existence(field_name):
+    """Return an Expression that is True iff `field_name` does not exist."""
+    local_field = LocalField(field_name)
+    local_field_size = UnaryTransformation(u'size', local_field)
+    field_null_check = BinaryComposition(u'=', local_field, NullLiteral)
+    field_size_check = BinaryComposition(u'=', local_field_size, ZeroLiteral)
+    return BinaryComposition(u'||', field_null_check, field_size_check)
+
+
+def _prune_traverse_using_omitted_locations(traverse,
+                                            omitted_locations,
+                                            optional_locations,
+                                            location_to_optional):
+    """Return a prefix of the given traverse, excluding any blocks after an omitted optional."""
+    new_traverse = []
+    for step in traverse:
+        if isinstance(step.root_block, Traverse) and step.root_block.optional:
+            in_optional_location = location_to_optional.get(
```

Comment:
this is pretty hard to read -- extract `step.as_block.location` to a variable and then use the variable instead